src/Instagram-Basic/Provider.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace SocialiteProviders\InstagramBasic;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'INSTAGRAMBASIC';
+
+
+    /**
+     * The user fields being requested.
+     *
+     * @var array
+     */
+    protected $fields = ['account_type', 'id', 'media_count', 'username'];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['user_profile', 'user_media'];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase(
+            'https://api.instagram.com/oauth/authorize',
+            $state
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://api.instagram.com/oauth/access_token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $meUrl = 'https://graph.instagram.com/me?access_token=' . $token . '&fields=' . implode(',', $this->fields);
+
+        if (!empty($this->clientSecret)) {
+            $appSecretProof = hash_hmac('sha256', $token, $this->clientSecret);
+            $meUrl .= '&appsecret_proof=' . $appSecretProof;
+        }
+        $response = $this->getHttpClient()->get($meUrl, [
+            'headers' => [
+                'Accept' => 'application/json',
+            ],
+        ]);
+
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id' => $user['id'],
+            'nickname' => $user['username'],
+            'name' => null,
+            'email' => null,
+            'avatar' => null,
+            'account_type' => $user['account_type'],
```

Comment:
Maybe worth using `Arr::get()` in case of upstream changes

Reply:
Not really sure why `Arr::get()`. Do you mean if the key `account_type` has been changed that we get null instead of an error?